cpp/core/jni/JniWrapper.cc

```diff
@@ -673,23 +685,21 @@ Java_io_glutenproject_columnarbatch_ColumnarBatchJniWrapper_getNumRows(JNIEnv* e
   JNI_METHOD_END(-1L)
 }
 
-JNIEXPORT jlong JNICALL Java_io_glutenproject_columnarbatch_ColumnarBatchJniWrapper_addColumn( // NOLINT
+JNIEXPORT jlong JNICALL Java_io_glutenproject_columnarbatch_ColumnarBatchJniWrapper_addIntColumn( // NOLINT
     JNIEnv* env,
     jobject,
     jlong handle,
     jint index,
-    jlong colHandle) {
+    jstring name,
+    jintArray colData) {
   JNI_METHOD_START
   std::shared_ptr<ColumnarBatch> batch = glutenColumnarBatchHolder.lookup(handle);
-  std::shared_ptr<ColumnarBatch> col = glutenColumnarBatchHolder.lookup(colHandle);
-#ifdef DEBUG
-  if (col->GetNumColumns() != 1) {
-    throw GlutenException("Add column should add one col");
-  }
-#endif
-  auto newBatch = batch->addColumn(index, col);
+  int32_t size = env->GetArrayLength(colData);
+  jint* col = env->GetIntArrayElements(colData, JNI_FALSE);
```

Comment:
We'd call `env->Release...` to release the column after used. Otherwise memory leak takes place.

Reply:
Yes, I have called after addIntColumn